### PR TITLE
feat: add sidenavigation partial action test

### PR DESCRIPTION
**Pull request by Samantha-KY**
# Submit a pull request

- [x] This is not a duplicate of an existing pull request.
- [x] No existing features have been broken without good reason.
- [x] Your commit messages are detailed
- [x] The code style guideline have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [x] Tests have been added or updated to reflect your changes.
- [x] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

This PR adds test for the SideNavigation component

:question:

## Breaking Changes

Describe what features are broken by this pull request and why, if any.

:question:

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

1. :question:

## Other Relevant Information

Provide any other important details below.

### CODE REVIEW

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    render(
+      <ActivableLink
+        item={{
+          subitems: [{ label: "subitem", link: "subitem link" }],
+          link: "/dummy",
+        }}
+      >
+        Link test
+      </ActivableLink>
+    );
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:20]: Can you create a rendern function for this component, you can refer to the button tests

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          link: "/dummy",
+        }}
+        isActive
+      >
+        Link test
+      </ActivableLink>
+    );
+    const link = screen.getByTestId("activableLinkId");
+    expect(link.children[0].tagName).toBe("SPAN");
+    expect(link.children[0].className).toContain("activable-link");
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:47]: let's avoid testing classes, the name of this call can change while the component is still behaving the same way.

File: __tests__/components/ui/SideNavigation/_partials/Content.test.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    render(<NavItem item={item} />);
+    const content = screen.getByTestId("contentId");
+    expect(content).toBeInTheDocument();
+  });
+
+  it("should render style when active", () => {
+    render(<NavItem item={item} isActive />);
+    const content = screen.getByTestId("contentId");
+    expect(content.children[0]).toHaveAttribute("style", "display: !inline-block;");
+  });
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Content.test.tsx:20]: This is hard to follow, let's avoid testing styles, the active style can change at anytime while the component still have the same behavour

File: __tests__/components/ui/SideNavigation/_partials/SideNavLink.test.tsx
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/SideNavLink.test.tsx:1]: Please format this file

File: src/components/ui/SideNavigation/_partials/Sub.tsx [hunk 4]
```diff
@@ -34,8 +34,8 @@ interface subLinkProps {
  */
 export default function SubLink({ item, activeLinkStyle, subitem }: subLinkProps): ReactElement {
   return (
-    <li className="relative mt-4 text-sm text-gray-500" style={activeLinkStyle}>
-      <Link href={{ pathname: item.link, hash: subitem.link }} className={`relative text-gray-500 opacity-50 hover:opacity-100 ${!subitem.exact ? "activable-link" : ""}`}>
+    <li data-testid="subId" className="relative mt-4 text-sm text-gray-500" style={activeLinkStyle}>
+      <Link data-testid="linkId" href={{ pathname: item.link, hash: subitem.link }} className={`relative text-gray-500 opacity-50 hover:opacity-100 ${!subitem.exact ? "activable-link" : ""}`}>
```
Comment (Jonath-z) [inline src/components/ui/SideNavigation/_partials/Sub.tsx:38]: I wonder, don't we have this already tested? because if it's tested then we don't need to test it again.

**Review by Jonath-z — CHANGES_REQUESTED** [review]
Consider also to format all the files please :)

File: __tests__/components/ui/SideNavigation/_partials/SubLink.test.tsx
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/SubLink.test.tsx:1]: Can you format this file please? Try to have prettier setup with your editor

File: __tests__/components/ui/SideNavigation/_partials/SubLink.test.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        render(<SubLink activeLinkStyle={{}} item={item} subitem={subitem}/>)
+        const sublink = screen.getByTestId("subLinkId")
+        expect(sublink).toBeInTheDocument()
+    })
+
+    it("should render sublink label", () => {
+        render(<SubLink item={item} activeLinkStyle={{}} subitem={subitem}/>)
+        const link = screen.getByText(subitem.label)
+        expect(link).toBeInTheDocument()
+        expect(link.firstChild?.textContent).toBe(subitem.label)
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/SubLink.test.tsx:35]: You can change the naming here, lable not link

File: __tests__/components/ui/SideNavigation/_partials/Sub.test.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        expect(sublink).toBeInTheDocument()
+    })
+
+    it("should render sub link label", () => {
+        render(<SubLink item={item} activeLinkStyle={{}} subitem={subitem}/>)
+        const link = screen.getByText(subitem.label)
+        expect(link).toBeInTheDocument()
+        expect(link.firstChild?.textContent).toBe(subitem.label)
+    })
+})
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Sub.test.tsx:29]: Here as well, it's lable and not link

**Review by Jonath-z — CHANGES_REQUESTED** [review]

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    expect(link).toBeInTheDocument();
+    expect(link.textContent).toBe("Link test");
+  });
+
+  it("should render the link with no subitems", () => {
+    render(<ActivableLink item={Emptyitem}>Link test</ActivableLink>);
+    const link = screen.getByText("Link test");
+    expect(link.tagName).toBe("A");
+    expect(link).toHaveAttribute("href", Emptyitem.link);
+  });
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:27]: You already tested the items, I don't think you need to test the empty items case

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx (hunk 2 above)
Comment (Jonath-z) [reply __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:37]: Still have this

File: __tests__/components/ui/SideNavigation/_partials/LinkAction.test.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        Link test
+      </LinkAction>
+    );
+    const link = screen.getByTestId("linkActionId");
+    expect(link).toBeInTheDocument();
+    expect(link.textContent).toBe("Link test");
+    fireEvent.click(link.children[0]);
+    expect(goToLinkMock).toHaveBeenCalled();
+  });
+});
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/LinkAction.test.tsx:31]: Can check that the number of subItems is rendered as the same number of component

File: __tests__/components/ui/SideNavigation/index.test.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+      <ReduxProvider>
+        <SideNavigation items={mockItems} colors={colors}>
+          <>Test</>
+        </SideNavigation>
+      </ReduxProvider>
+    );
+    const navItems = screen.getAllByTestId("sideNavId");
+    expect(navItems.length).toBe(1);
+  });
+});
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/index.test.tsx:80]: you can also check the items number that are in the mock

**Review by Jonath-z — CHANGES_REQUESTED** [review]

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    render(
+      <ActivableLink item={items} isActive>
+        Link test
+      </ActivableLink>
+    );
+    const link = screen.getByTestId("activableLinkId");
+    expect(link.children[0].tagName).toBe("SPAN");
+  });
+
+  it("should call goToLink function when clicked", () => {
```
Comment (serapieTuyishime) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:28]:
```suggestion
  it("should invoke goToLink function when clicked", () => {
```

**Review by Jonath-z — APPROVED** [review]

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+  it("should render the link with activable link style", () => {
+    render(
+      <ActivableLink item={items} isActive>
+        Link test
+      </ActivableLink>
+    );
+    const link = screen.getByTestId("activableLinkId");
+    expect(link.children[0].tagName).toBe("SPAN");
+  });
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:25]: The test description does not explain the code tested here,  would you check this again?

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+  it("should invoke goToLink function when clicked", () => {
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:27]:
```suggestion
  it("should redirect to link when clicked", () => {
```

File: src/config/firebase.ts [hunk 13]
```diff
@@ -8,11 +8,11 @@ let analytics: Analytics;
...
-    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
-    projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
-    authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
-    appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
-    measurementId: process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID,
+    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY || "mock-key",
+    projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID || "mock-project",
+    authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN || "mock-auth",
+    appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID || "mock-app",
+    measurementId: process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID || "mock-measurement",
```
Comment (Jonath-z) [inline src/config/firebase.ts:15]:
```suggestion
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY ?? " ",
    projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID ?? " ",
    authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN ?? " ",
    appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID ?? " ",
    measurementId: process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID ?? " ",
```

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+  subitems: [{ label: "subitem", link: "subitem link" }],
+  link: "/dummy",
+};
+describe("Activable link", () => {
+  it("should render the link with subitems", () => {
+    render(<ActivableLink item={items}>Link test</ActivableLink>);
+    const link = screen.getByTestId("activableLinkId");
+    expect(link).toBeInTheDocument();
+    expect(link.textContent).toBe("Link test");
+  });
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:15]: Would you test with many subitems as well?

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+  it("should render the link with active status", () => {
+    render(
+      <ActivableLink item={items} isActive>
+        Link test
+      </ActivableLink>
+    );
+    const link = screen.getByTestId("activableLinkId");
+    expect(link.children[0].tagName).toBe("SPAN");
+  });
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:25]: how does this attest the the link is active?

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+
+  it("should redirect to link when clicked", () => {
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:27]:
```suggestion
  it("should invok goToLink when clicked", () => {
```

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+import "@testing-library/jest-dom";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+const items = {
+  subitems: [{ label: "subitem", link: "subitem link" }],
+  link: "/dummy",
+};
+describe("Activable link", () => {
+
+  it("should render the link with many subitems", () => {
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:11]:
```suggestion
  it("should render the link when subitems are provided", () => {
```

File: __tests__/components/ui/SideNavigation/_partials/Action.test.tsx [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+    const itemsWithManySubitems = {
+      ...items,
+      subitems: [
+        { label: "subitem1", link: "subitem-link-1" },
+        { label: "subitem2", link: "subitem-link-2" },
+      ],
+    };
```
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/_partials/Action.test.tsx:18]: You can make this one mock with items, no need to separate them

**Review by Jonath-z — COMMENTED** [review]
You can test the LinkAction component the same way you tested the ActivableLink component as they have the same structure and same logic

File: __tests__/components/ui/SideNavigation/index.test.tsx
Comment (Jonath-z) [inline __tests__/components/ui/SideNavigation/index.test.tsx:1]: There still some points to test in this component, for example the menu title  when it's hidden or it's not, we need to test the case where we don't have any item passed, or we have the item but not the subItem
